libs/effects-ng/src/lib/directive-effects.ts

```diff
@@ -0,0 +1,59 @@
+import { Directive, Type, inject, OnDestroy } from '@angular/core';
+import { EFFECTS_MANAGER } from './tokens';
+import { getEffectProps } from './utils';
+import { Subject } from 'rxjs';
+import { Effect } from '@ngneat/effects';
+
+@Directive()
+export abstract class EffectsDirective implements OnDestroy {
+  protected readonly manager = inject(EFFECTS_MANAGER, { optional: true });
+  protected destroySubject = new Subject<void>();
+
+  protected constructor() {
+    if (!this.manager) {
+      throw new TypeError(
+        "Effects manager wasn't found. Have you forgotten to provide it? Please call `provideEffectsManager` in the root providers."
+      );
+    }
+  }
+
+  ngOnDestroy(): void {
+    this.destroySubject.next();
+    this.destroySubject.complete();
+  }
+}
+
+export function directiveEffects(
```

Comment:
```suggestion
export function useEffectsDirective(
```

Reply:
It sounds kind of `React`ly. But... don't have any other names.